### PR TITLE
[Merged by Bors] - feat: port Init.Data.Prod

### PR DESCRIPTION
**Pull request by pechersky**
add `#align prod Prod` to Init.Data.Core

should the align be in Init.Data.Prod instead?

### CODE REVIEW

File: Mathlib/Init/Data/Prod.lean [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/-
+Copyright (c) 2014 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Leonardo de Moura, Jeremy Avigad
+-/
+prelude
```
Comment (dwrensha) [inline Mathlib/Init/Data/Prod.lean:6]: Is there a particular reason to include this `prelude` line? None of the other ported files in `Mathlib/Init` include it. (I see that the mathport output includes this line, but I'm not confident that's intentional.)

Comment (pechersky) [reply Mathlib/Init/Data/Prod.lean:6]: I am fine removing this and not having to rely on the style linter change. Having some explicit guidelines is helpful for me.

Comment (pechersky) [reply Mathlib/Init/Data/Prod.lean:6]: In Lean3 init.data.prod:
```
prelude
import init.logic
universes u v
```
In Lean4 Init.Data.Prod:
```
prelude
import Init.SimpLemmas
``

Comment (digama0) [reply Mathlib/Init/Data/Prod.lean:6]: You should strip the `prelude`. You should generally only take the mathlib3port import lines as a guideline and adjust it to match mathlib4 / lean4 anyway. Mathport does not have a very nuanced understanding of what imports are actually necessary and is just transcribing what the lean 3 file said, but mathlib4 will not have any `prelude` files. Being `prelude` in lean 3 just means that you are not importing `Mathlib.Init` which is now a regular import.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**